cql3/statements/modification_statement.cc

```diff
@@ -414,7 +419,22 @@ modification_statement::execute_without_condition(service::storage_proxy& proxy,
             return now();
         }
 
-        return proxy.mutate_with_triggers(std::move(mutations), cl, timeout, false, qs.get_trace_state(), qs.get_permit(), this->is_raw_counter_shard_write());
+        auto mutate = [&, cl, timeout] (std::vector<mutation> mutations) {
+            return proxy.mutate_with_triggers(std::move(mutations), cl, timeout, false, qs.get_trace_state(), qs.get_permit(), this->is_raw_counter_shard_write());
+        };
+
+        if (qs.get_client_state().is_internal() || qs.get_client_state().is_thrift() || !s->cdc_enabled()) {
+            return mutate(std::move(mutations));
+        }
+        return cdc::append_log_mutations(
```

Comment:
I don't remember, why does this append_log_mutations (which just adds more mutations to the vector) return a future instead of working immediately?

Reply:
It reads the CDC description table to create the mutations.
I'm implementing a topology-change-proof version which won't read the description table, but we will still have to read the base table to create the pre-image, so it will eventually have to return a future.